### PR TITLE
Sort object detection results and reduce code duplication

### PR DESCRIPTION
**Pull request by mcm001**

### CODE REVIEW

File: photon-client/src/components/dashboard/tabs/ObjectDetectionTab.vue [hunk 1]
```diff
@@ -32,5 +42,42 @@ const interactiveCols = computed(() =>
...
+      @input="(value) => useCameraSettingsStore().changeCurrentPipelineSetting({ contourArea: value }, false)"
+    />
+    <pv-range-slider
+      v-model="contourRatio"
+      label="Ratio (W/H)"
+      tooltip="Min and max ratio between the width and height of a contour's bounding rectangle"
+      :min="0"
+      :max="100"
+      :slider-cols="interactiveCols"
+      :step="0.1"
```
Comment (mdurrani808) [inline photon-client/src/components/dashboard/tabs/ObjectDetectionTab.vue:61]: is there a reason for these having different step sizes

File: photon-core/src/main/java/org/photonvision/vision/opencv/Contour.java [hunk 2]
```diff
@@ -47,6 +47,16 @@ public Contour(MatOfPoint mat) {
         this.mat = mat;
     }
 
+    public Contour(Rect2d box) {
+        // no easy way to convert a Rect2d to Mat, diy it. Order is tl tr br bl
```
Comment (mdurrani808) [inline photon-core/src/main/java/org/photonvision/vision/opencv/Contour.java:51]: https://docs.opencv.org/4.x/javadoc/org/opencv/core/MatOfRect2d.html

This relevant?

File: photon-core/src/main/java/org/photonvision/vision/pipeline/ObjectDetectionPipeline.java [hunk 3]
```diff
@@ -52,6 +57,38 @@ protected void setPipeParamsImpl() {
...
+                new DualOffsetValues(
+                        settings.offsetDualPointA,
+                        settings.offsetDualPointAArea,
+                        settings.offsetDualPointB,
+                        settings.offsetDualPointBArea);
+
+        SortContoursPipe.SortContoursParams sortContoursParams =
+                new SortContoursPipe.SortContoursParams(
+                        settings.contourSortMode,
+                        settings.outputShowMultipleTargets ? 5 : 1,
```
Comment (mdurrani808) [inline photon-core/src/main/java/org/photonvision/vision/pipeline/ObjectDetectionPipeline.java:71]: what are these magic numbers

**Review by mdurrani808 — COMMENTED** [review]
Didn't test so won't approve but once someone does lgtm

File: photon-core/src/main/java/org/photonvision/vision/pipeline/ObjectDetectionPipeline.java (hunk 3 above)
Comment (mcm001) [reply photon-core/src/main/java/org/photonvision/vision/pipeline/ObjectDetectionPipeline.java:71]: Duplicated from other pipelines, but max targets in multi target mode.'I'll document

File: photon-core/src/main/java/org/photonvision/vision/target/PotentialTarget.java [hunk 4]
```diff
@@ -41,12 +47,26 @@ public PotentialTarget(Contour inputContour, List<Contour> subContours, CVShape
         m_mainContour = inputContour;
         m_subContours = new ArrayList<>(subContours);
         this.shape = shape;
+        this.clsId = -1;
```
Comment (mdurrani808) [inline photon-core/src/main/java/org/photonvision/vision/target/PotentialTarget.java:50]: we use like 3 different names for class id but i dont really care all too much

**Review by mdurrani808 — APPROVED** [review]

**Pull request merged**